server/modules/navigator/navigator.go

```diff
@@ -0,0 +1,370 @@
+				baseTechnique := strings.ToUpper(strings.Split(match, ".")[0])
+				techniques[baseTechnique] = struct{}{}
+			}
+		}
+	}
+
+	return techniques
+}
+
+// extractAlertTechniques extracts technique IDs from alerts in the last N days
+func (nav *Navigator) extractAlertTechniques(ctx context.Context, logger *log.Entry) techniqueMap {
+	// Create a context with timeout
+	ctx, cancel := context.WithTimeout(ctx, ALERT_EXTRACTION_TIMEOUT)
+	defer cancel()
+
+	techniques := make(techniqueMap)
+
+	// Query alerts from the last N days
+	lookback := time.Duration(nav.lookbackDays) * 24 * time.Hour
+	startTime := time.Now().Add(-lookback)
+
+	// Build the search criteria
+	criteria := model.NewEventSearchCriteria()
+	criteria.RawQuery = "*"
+	criteria.BeginTime = startTime
+	criteria.EndTime = time.Now()
+	criteria.EventLimit = 10000 // Keep within Elasticsearch limit
+	criteria.SortFields = []*model.SortCriteria{{Field: "@timestamp", Order: "desc"}}
+	criteria.SearchAfter = []interface{}{time.Now().UnixNano() / int64(time.Millisecond)}
+
+	// Create the query to filter by index
+	if err := criteria.ParsedQuery.Parse("event.module:suricata AND event.dataset:suricata.alert"); err != nil {
+		logger.WithError(err).Error("failed to parse query")
+		return techniques
+	}
+
+	logger.WithFields(log.Fields{
+		"rawQuery":   criteria.RawQuery,
+		"beginTime":  criteria.BeginTime.Format(time.RFC3339),
+		"endTime":    criteria.EndTime.Format(time.RFC3339),
+		"eventLimit": criteria.EventLimit,
+	}).Info("Executing Suricata alert search query")
+
+	var totalProcessed int
+
+	for {
+		select {
+		case <-ctx.Done():
+			if ctx.Err() == context.DeadlineExceeded {
+				logger.WithField("processed_events", totalProcessed).Warn("alert extraction timed out after 3 minutes")
+			}
+			return techniques
+		default:
+			// Execute the search
+			searchResult, err := nav.server.Eventstore.Search(ctx, criteria)
+			if err != nil {
+				logger.WithError(err).Error("failed to search alerts")
+				return techniques
+			}
+
+			logger.WithFields(log.Fields{
+				"totalEvents": searchResult.TotalEvents,
+				"batchSize":   len(searchResult.Events),
+				"processed":   totalProcessed,
+			}).Info("Processing batch of Suricata alerts for navigator layer")
+
+			for _, event := range searchResult.Events {
+				techniqueIDs, ok := event.Payload["rule.metadata.mitre_technique_id"].([]interface{})
+				if !ok {
+					continue
+				}
+
+				for _, id := range techniqueIDs {
+					if techID, ok := id.(string); ok {
+						techniques[strings.ToUpper(techID)] = struct{}{}
+					}
+				}
+			}
+
+			totalProcessed += len(searchResult.Events)
+
+			// Check if we've processed all events or got an empty batch
+			if totalProcessed >= searchResult.TotalEvents || len(searchResult.Events) == 0 {
+				return techniques
+			}
+
+			// Get the sort values from the last event for the next search
+			lastEvent := searchResult.Events[len(searchResult.Events)-1]
+			criteria.SearchAfter = lastEvent.Sort
```

Comment:
Is the `Sort` field the correct value to use for pagination? I haven't looked at the ES API in a while so it might be. Just seems odd to use a sort term for pagination.